### ISSUE TITLE
[Bug]: Instant GPU Switching doesn't work on MacBookPro5,2 due to wrong AppleMuxControl.kext version

### ISSUE THREAD

**Issue by HackintoshHD** [dortania/OpenCore-Legacy-Patcher#717]
### Machine Model

MacBookPro5,2

### Application Version

Latest Build (nightly)

### Application Variant

TUI (Text User Interface)

### Have you verified whether this issue has been opened before? If no, your issue will be closed

true

### Have you verified whether this issue is covered in our guide? ie. Troubleshooting and Legacy Acceleration pages

true

### Is this issue with a 3rd party application?

false

### What versions of macOS are you seeing the problem on?

macOS 11, Big Sur

### Where does this issue happen?

Within macOS (After Logging in)

### What is the Issue?

First of all, let me thank the OCLP developers for their brilliant work which is really appreciated.

According to the OCLP 0.3.1 changlog entry, Instant GPU Switching / GPU Switching on the Fly has been added:

> Update Legacy GMUX patchset to latest Sierra Secuirty Update
>
> [Source](https://github.com/HackintoshHD/mbp5x-instant-gpu-switching)

On my MacBook Pro Mid 2009 (MacBookPro5,2) however, neither OCLP 0.3.1 TUI nor today's OCLP 0.3.2 TUI nightly deploy the necessary `AppleMuxControl.kext` version 3.14.52 when applying the Post-Install Volume Patch. On my root volume-patched macOS 11.6.1 installation, `/System/Library/Extensions/AppleGraphicsControl.kext/Contents/PlugIns/AppleMuxControl.kext` is clearly version 6.3.5 and switching between the integrated Nvidia GeForce 9400M and the discrete Nvidia GeForce 9600M GT via [gfxCardStatus 2.5](https://github.com/codykrieger/gfxCardStatus) therefore fails.

Hence my request: Would it be possible to add an option in OCLP's patcher settings (currently menu entry no. 5) for the MacBookPro5,2, e.g. something like

> 8. Enable Instant GPU Switching on MBP5,x (disables Brightness Control)

so, if activated, OCLP could install `AppleMuxControl.kext` version 3.14.52 and enable Instant GPU Switching?

Maybe such a switch would be the best option as some users (cc: @hvdschmitt) prefer to keep brightness control while some users (including me) prefer to be able to switch to the internal GPU to run the machine significantly cooler than with dedicated graphics.

Thank you very much!

### Any Additional Information

Follow-up issue to #514 and #584.

**Comment by khronokernel** [comment]
GMUX patch set is disabled here until someone wishes to further validate:

* https://github.com/dortania/OpenCore-Legacy-Patcher/blob/main/resources/sys_patch.py#L763-L764

Uncomment the above model check, comment out below model check.

As we've already had multiple users file issues with concerns. If you validate that this code change is in fact beneficial and not harmful, feel free to make a PR to the project. However as of right now, do not have time to continue maintaining this feature request.

* If you validate it works and assume responsibility for the patch set, I will update your submitted PR with the CLI menu option to toggle such setting. Don't need to worry about writing python for the PR outside the line comment/uncomment I mentioned earlier

**Comment by HackintoshHD** [comment]
@khronokernel Thank you very much for the explanation. I've had a look at the relevant lines and realize it's basically undoing commit 6bc36c737c228830d69b1acfa444a5d0105d5af8. I'll try to have a closer look first at how OCLP activated Instant GPU Switching on MBP5,x before it was deactivated again by the referenced commit. Would it be possible to leave this issue open?

Thanks again!

**Comment by HackintoshHD** [comment]
Okay, I think I've understood the code. As mentioned in https://github.com/dortania/OpenCore-Legacy-Patcher/issues/717#issue-1049732617 above, simply reactivating it would mean loosing backlight control, and that choice "Instant GPU Switching vs. Backlight Control" should IMHO be left to the user. So I'd need to add another variable in parallel to the patch settings. I'll check and see what I can do with a relatively small time budget.
